api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -339,6 +355,9 @@ message Watchdog {
   // nonresponsive threads required for the *multikill_timeout*.
   // If not specified the default is 0.
   type.v3.Percent multikill_threshold = 5;
+
+  // Register actions that will fire on given WatchDog events.
```

Comment:
you may want to say something about any guarantees about ordering that you intend to provide.  They are mostly relevant for the case of KILL/MULTIKILL where it may be useful to specify several debug related actions followed by an alternate FATAL action used to replace the default guarddog initiated fatal action.